arcgis-ios-sdk-samples/Content Display Logic/Model/Sample.swift

```diff
@@ -21,6 +21,23 @@ struct Sample: Hashable {
     var description: String
     var storyboardName: String
     var dependencies: [String]
+    var isFavorite: Bool {
```

Comment:
Please move this into an extension to keep from cluttering the type. Also, think about this. The value stored in the dictionary is always `true`, which means that the value isn't needed. The presence of the key/value pair is enough to indicate whether the sample is a favorite. So what we really only need is an unordered collection of unique keys. What does that sound like? A set. `UserDefaults ` doesn't support sets, but it does support arrays. The storage should be an array of sample names viewed as a set. Since that set is really a property of the user defaults instance, I would recommend modeling it as such:

```swift
extension UserDefaults {
    static let favoriteSamplesKey = "favoriteSamples"
    
    var favoriteSamples: Set<String> {
        get {
            Set(stringArray(forKey: Self.favoriteSamplesKey)!)
        }
        set {
            set(newValue.sorted(), forKey: Self.favoriteSamplesKey)
        }
    }
}
```

And then the implementation of `isFavorite` would look something like this:

```swift
extension Sample {
    var isFavorite: Bool {
        get {
            UserDefaults.standard.favoriteSamples.contains(name)
        }
        nonmutating set {
            if newValue {
                UserDefaults.standard.favoriteSamples.insert(name)
            } else {
                UserDefaults.standard.favoriteSamples.remove(name)
            }
        }
    }
}
```